mlrun/platforms/iguazio.py

```diff
@@ -424,6 +423,54 @@ def dump_record(rec):
             )
 
 
+class KafkaOutputStream:
+    def __init__(
+        self,
+        topic,
+        brokers,
+        mock=False,
+    ):
+        self._kafka_producer = None
+        self._topic = topic
+        self._brokers = brokers
+
+        self._mock = mock
+        self._mock_queue = []
+
+        self._initialized = False
+
+    def _lazy_init(self):
+        if self._initialized:
+            return
+
+        import kafka
+
+        self._kafka_producer = kafka.KafkaProducer(
+            bootstrap_servers=self._brokers,
+        )
+
+        self._initialized = True
+
+    def push(self, data):
+        self._lazy_init()
+
+        def dump_record(rec):
+            if not isinstance(rec, (str, bytes)):
+                return dict_to_json(rec)
+            return str(rec).encode("UTF-8")
```

Comment:
Not clear why this is the behavior here - 
* If data was bytes, no need to do `str()` and then re-encode it. 
* Why is the data converted to `json` if its str/bytes, and not encoded to bytes?